### DISCUSSION TITLE
Cypress 10 attempting version check behind company firewall.

### DISCUSSION THREAD

**Discussion by kc2112** [cypress-io/cypress#22346] [General]
Good day!!  Cypress 10 does a version check when it starts a testing mode.  My company is behind a firewall and will not allow the call to be completed.  As a result Cy 10 crashes.  How can we disable that check so we can still use the library??  Line 125 is failing due to invalid json response body when trying to parse https://download.cypress.io/desktop.json (which we cannot download)...

I was able to bypass the check by adding a return statement on line 121 of the following file:

C:\Users\XXXXX\AppData\Local\Cypress\Cache\10.0.0\Cypress\resources\app\packages\data-context\src\sources\VersionDataSource.js
return pkg.version; //on line 121

Can we get this check made optional, or at least configured to where it doesn't fail when the check cannot be made??  Or possibly validate the response object before operations are attempted on it.  Something to fix, please!!

**Comment by Glacian22** [comment]
Does your company have a proxy to let you through the firewall? If so, I'd check out the [doc on proxy configuration](https://docs.cypress.io/guides/references/proxy-configuration).

**Reply by kc2112** [reply]
They will not allow that, no. The application should not fail on something so simple as not validating a response object.
